process.py

```diff
@@ -224,57 +248,61 @@ def add_biomarkers(elements, max_depth, is_reference):
                     error += "\n\tFeature: " + element
                     exit_with_error(error)
             if is_reference:
-                output += get_reference_block(features[element]) + "\t"
+                referenceList=get_reference_block(features[element])
+                if referenceList:
+                    for references in referenceList:
+                        output.append(references)
             else:
-                output += get_data_block(features[element]) + "\t"
+                dataBlockList=get_data_block(features[element])
+                if dataBlockList:
+                    for dataBlock in dataBlockList:
+                        output.append(dataBlock)
             count += 1
     if count < max_depth:
-        output += "\t\t\t\t\t" * (max_depth-count)
+        for m in range((max_depth-count)):
+            for n in range(5):
+                output.append("")
     return output
 
-
-def add_features(record, element):
+def add_features(record:list, element)->None:
     """
     The argument (element) can be either an anatomical structure ("node")
     or a cell as they contain the same features.
     """
-    global features, out_file
-
     # need to track how many entities we output per feature type, to
     # appropriately pad the output.
-    count = 0
-
+    
     # add biomarkers
-    record += add_biomarkers(element.genes, max_genes, False)
-    record += add_biomarkers(element.proteins, max_proteins, False)
-    record += add_biomarkers(element.proteoforms, max_proteoforms, False)
-    record += add_biomarkers(element.lipids, max_lipids, False)
-    record += add_biomarkers(element.metabolites, max_metabolites, False)
-    record += add_biomarkers(element.ftu, max_ftu, False)
-    record += add_biomarkers(element.references, max_references, True)
-
-    record += "\n"
-    out_file.write(record)
-
-
-def print_ASCTB_table():
+    allBiomarkers=[]
```

Comment:
For these where its var=[] and then you append right afterward, you can rewrite to initialize the array in one go like this;
var=[
  add_biomarkers(...),
  add_biomarkers(...)
  ...
]